### PR TITLE
musl fixes

### PR DESCRIPTION
**Pull request by kraj**

### CODE REVIEW

**Review by otavio — APPROVED** [review]

File: recipes-graphics/imx-g2d/imx-dpu-g2d_2.1.0.bb [hunk 1]
```diff
@@ -22,6 +22,10 @@ do_install () {
     cp -Pr ${S}/g2d/usr/include/* ${D}${includedir}
 }
 
+RDEPENDS:${PN}:append:libc-musl = " gcompat"
+
```
Comment (otavio) [inline recipes-graphics/imx-g2d/imx-dpu-g2d_2.1.0.bb:26]: Please make sure this is consistent on every recipe (close to the end of the recipe and without newlines).

**Review by otavio — CHANGES_REQUESTED** [review]

**Comment by kraj** [comment]
Cool, now I have `yoe-ptest-image` with weston running with musl on a `imx8qm-variscite-som` box

**Review by otavio — APPROVED** [review]

File: recipes-graphics/xwayland/xwayland_%.bbappend [hunk 2]
```diff
@@ -10,3 +10,10 @@ OPENGL_PKGCONFIGS_REMOVE_IMXGPU             = ""
 OPENGL_PKGCONFIGS_REMOVE_IMXGPU:mx6-nxp-bsp = "glamor glx"
 OPENGL_PKGCONFIGS_REMOVE_IMXGPU:mx7-nxp-bsp = "glamor glx"
 OPENGL_PKGCONFIGS_REMOVE_IMXGPU:mx8-nxp-bsp = "glx"
+
+
+# links with imx-gpu libs which are pre-built for glibc
+# gcompat will address it during runtime
+LDFLAGS:append:imxgpu3d:libc-musl = " -Wl,--allow-shlib-undefined"
+LDFLAGS:append:imxgpu:libc-musl = " -Wl,--allow-shlib-undefined"
```
[the remark this answers is not in the document]
Comment (kraj) [reply recipes-graphics/xwayland/xwayland_%.bbappend:18]: Are there machines which are just imxgpu and not 3D ? The second one covers those

[the remark this answers is not in the document]
Comment (thochstein) [reply recipes-graphics/xwayland/xwayland_%.bbappend:18]: Yes, they are `imxgpu:imxgpu2d`. So `imxgpu` override is sufficient and you can drop the `imxgpu3d` override.

**Review by otavio — APPROVED** [review]

**Review by thochstein — APPROVED** [review]

**Review by angolini — APPROVED** [review]

**Pull request merged**